cranelift/isle/isle/src/sema.rs

```diff
@@ -80,9 +80,36 @@ pub struct TypeEnv {
     pub errors: Vec<Error>,
 }
 
+/// A built-in type.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum BuiltinType {
```

Comment:
I'm not sure I understand the reason for having a notion of "built-in types" for bool but not for other types that we have constants for (namely integers) -- while on the other hand it adds some complication to the IR. Maybe we could separate out this part into another PR and discuss separately?

Reply:
(To make it more explicit: I think we probably *should* have this eventually; but for integers too, and then we should check that integer literals are used only in contexts that expect them.)

Reply:
The intention is to extend `BuiltinType` to include other types (like `u{8,16,32,64,size}, `i{8,16,32,64,size}` or `str`) in future PRs

Reply:
OK, yeah, on further thought I suppose it's totally fine to migrate integers over as a second step -- as long as the intention is there to finish out the refactor. Thanks!